ipaserver/plugins/baseuser.py

```diff
@@ -648,6 +648,15 @@ def check_namelength(self, ldap, **options):
                         )
                     )
 
+    def check_name(self, entry_attrs):
+        if 'uid' in entry_attrs:
+            # Check the pattern if the user is renamed
+            pattern = re.compile(constants.PATTERN_GROUPUSER_NAME)
```

Comment:
Otherwise the change checks out and looks good.

Reply:
> If you make pattern a class variable then it will be compiled once and can be re-used during the runtime. Otherwise you can skip the compile and just do the match.

Good suggestion, implemented.